src/dot.rs

```diff
@@ -1,64 +1,59 @@
 use std::io::Write;
 use std::borrow::Cow;
 
-use crate::vertex_id::VertexId;
 use crate::graph::Graph;
-use std::sync::Arc;
-use std::fmt::Debug;
+use std::fmt::Display;
 
-type Nd = isize;
-type Ed= (isize, isize);
-struct Edges(Vec<Ed>);
+/*
+Bounds on types throw warnings
+type Nd<D: Display + Clone + Ord> = D;
+type Ed<D: Display + Clone + Ord> = (D, D);
+*/
+type Nd<D> = D;
+type Ed<D> = (D, D);
 
-pub fn render_to(graph: &Graph<isize>, output: &mut impl Write) {
+struct Edges<D: Display + Clone>(Vec<Ed<D>>);
+
+pub fn render_to(graph: &Graph<impl Display + Clone + Ord>, output: &mut impl Write) {
```

Comment:
I think it is better if we move this to the graph structure as a method, something like `Graph::to_dot` and `Graph::from_dot` to keep the api simple.

This also makes it unnecessary to expose an `Graph::edges` method.

Reply:
I moved the `render_to` method to `Graph` and renamed it. However, I still need the `edges()` and  `hashmap_vertices` methods because `Graph::to_dot` method is still a static method and cannot be changed to `mygraph.to_dot()` due to the `Display + Clone + Ord` requirement.

Reply:
Ok, this makes sense.